android/build.gradle

```diff
@@ -154,6 +151,7 @@ def downloadsDir = customDownloadsDir ? new File(customDownloadsDir) : new File(
 def thirdPartyNdkDir = new File("$buildDir/third-party-ndk")
 
 def reactNativeThirdParty = new File("$reactNativeRootDir/ReactAndroid/src/main/jni/third-party")
+def reactNativeAndroidDownloadDir = new File("$reactNativeRootDir/ReactAndroid/build/downloads")
```

Comment:
It should work just fine in case the variable is set.

We also set reanimated's download directory with respect to whether `REACT_NATIVE_DOWNLOADS_DIR` env var is set or not [(link)](https://github.com/facebook/react-native/blob/596111fabf93e65c0a3880a4d6be56368a809ec5/ReactAndroid/build.gradle#L34-L35)
```groovy
def customDownloadsDir = System.getenv("REACT_NATIVE_DOWNLOADS_DIR")
def downloadsDir = customDownloadsDir ? new File(customDownloadsDir) : new File("$buildDir/downloads")
```

So in case the env variable is set, we use the same download directory as React Native, but our download tasks won't start before `:ReactAndroid`'s are over, as there is implicit (through `resolve<lib>` tasks) dependency between them --> they will terminate with `UP-TO-DATE` state.
`resolve<lib>` tasks terminate with `NO-SOURCE` state in case "from" directory does not exist (they do not throw any exception).

In case there is no `:ReactAndroid` project (Paper) it also should work fine.